Swiftification/NSAttributedString.swift

```diff
@@ -30,9 +30,8 @@ public func + (lhs: NSAttributedString, rhs: NSAttributedString) -> NSAttributed
 }
 
 public extension Sequence where Iterator.Element == NSAttributedString {
-
     /// Interpose the `separator` between elements of `self`, then concatenate the result.
-    func joinWithSeparator(_ separator: NSAttributedString) -> NSAttributedString {
+    func joined(separator: NSAttributedString) -> NSAttributedString {
```

Comment:
I went with `joined(separator:)` to match the existing function on `String` sequences (http://swiftdoc.org/v3.0/protocol/Sequence/#func-iterator-element-string-joined_).